### PR TITLE
Allow Sync Frequency to be nil

### PR DESCRIPTION
**Pull request by fivetran-jovanmanojlovic**
Related to [T-765145](https://fivetran.height.app/T-765145)

### CODE REVIEW

File: fivetran/framework/resources/connector_schedule.go [hunk 1]
```diff
@@ -49,8 +49,14 @@ func (r *connectorSchedule) Create(ctx context.Context, req resource.CreateReque
...
-		ConnectorID(data.ConnectorId.ValueString()).
-		SyncFrequency(helpers.StrToInt(data.SyncFrequency.ValueString()))
+		ConnectorID(data.ConnectorId.ValueString())
+
+	if data.SyncFrequency.ValueString() != "" {
+		syncFrequency := helpers.StrToInt(data.SyncFrequency.ValueString())
+		svc.SyncFrequency(&syncFrequency)
+	} else {
+		svc.SyncFrequency(nil)
+	}
```
Comment (beevital) [inline fivetran/framework/resources/connector_schedule.go:59]: We should check `if !data.SyncFrequency.IsNull() && !data.SyncFrequency.IsUnknown() || data.SyncFrequency.ValueString() != ""`

File: fivetran/framework/resources/connector_schedule.go [hunk 2]
```diff
@@ -142,7 +148,12 @@ func (r *connectorSchedule) Update(ctx context.Context, req resource.UpdateReque
 		ConnectorID(state.ConnectorId.ValueString())
 
 	if !plan.SyncFrequency.Equal(state.SyncFrequency) {
-		svc.SyncFrequency(helpers.StrToInt(plan.SyncFrequency.ValueString()))
+		if plan.SyncFrequency.ValueString() != "" {
```
Comment (beevital) [inline fivetran/framework/resources/connector_schedule.go:151]: Same here - we should check IsNull() and IsUnknown() as well.

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,7 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## [Unreleased](https://github.com/fivetran/terraform-provider-fivetran/compare/v1.2.8...HEAD)
+## [Unreleased](https://github.com/fivetran/terraform-provider-fivetran/compare/v1.2.9...HEAD)
+
+## [1.2.0](https://github.com/fivetran/terraform-provider-fivetran/compare/v1.2.8...v1.2.9)
+
+## Fixed
+- Issue with SyncFrequency unable to be set to nil
```
Comment (beevital) [inline CHANGELOG.md:13]:
```suggestion
- Issue with resource `fivetran_connector_schedule` when SyncFrequency is unknown.
```

We should introduce issues in change-log notes from user-understanding side. It's not clear what does this mean here, only we as a developers know that the issue caused by this fact. But from customer side it should be described in different manner.

**Review by beevital — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,7 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## [Unreleased](https://github.com/fivetran/terraform-provider-fivetran/compare/v1.2.8...HEAD)
+## [Unreleased](https://github.com/fivetran/terraform-provider-fivetran/compare/v1.2.9...HEAD)
+
+## [1.2.0](https://github.com/fivetran/terraform-provider-fivetran/compare/v1.2.8...v1.2.9)
```
Comment (fivetran-aleksandrboldyrev) [inline CHANGELOG.md:10]:
```suggestion
## [1.2.9](https://github.com/fivetran/terraform-provider-fivetran/compare/v1.2.8...v1.2.9)
```

**Review by fivetran-aleksandrboldyrev — APPROVED** [review]
LGTM

**Review by beevital — APPROVED** [review]

**Pull request merged**